main.tf

```diff
@@ -1 +1,234 @@
-# main.tf placeholder
+locals {
+  deny_leaving_orgs_effect             = var.deny_leaving_orgs ? "Deny" : "Allow"
+  deny_creating_iam_users_effect       = var.deny_creating_iam_users ? "Deny" : "Allow"
+  deny_deleting_kms_keys_effect        = var.deny_deleting_kms_keys ? "Deny" : "Allow"
+  deny_deleting_route53_zones_effect   = var.deny_deleting_route53_zones ? "Deny" : "Allow"
+  deny_deleting_cloudwatch_logs_effect = var.deny_deleting_cloudwatch_logs ? "Deny" : "Allow"
+  deny_root_account_effect             = var.deny_root_account ? ["Deny"] : []
+  protect_s3_buckets_effect            = var.protect_s3_buckets ? ["Deny"] : []
+  protect_iam_roles_effect             = var.protect_iam_roles ? ["Deny"] : []
+  limit_regions_effect                 = var.limit_regions ? ["Deny"] : []
+}
```

Comment:
Why are these lists?

Reply:
Funny story: AWS only acknowledges some of these policies as "Deny". When I tried to set the `effect` to either Deny or Allow, I get a malformed policy document error on "Allow". Using `dynamic` + `for_each` and then setting the `effect` to either "Deny" or an empty array prevents the "Allow" problem and lets the `effect` (and related variable switch) work. 

If it's stupid but it works...it ain't stupid?

Reply:
I guess I'm just not sure how a list is being interpreted in a field that expects either "Allow" or "Deny". That seems...weird.

Reply:
It is def a bit weird.

The `effect` field is only ever "Deny" for these. We're either including the "Deny" statement (with hard-coded Deny in the `content` param ) or passing no statement through at all. For naming, the Deny/(alternative) pattern just provides some consistency in the module call.